src/Storefront/Theme/ThemeMergedConfigBuilder.php

```diff
@@ -109,6 +109,8 @@ public function getPlainThemeConfiguration(string $themeId, Context $context): a
             }
         }
 
+        $themeConfig['themeTechnicalName'] = u($theme->getTechnicalName() ?? $theme->getName())->kebab();
```

Comment:
I would not prefer to use a magical kebab version for the technical name. Only database copies don't have a technical name, and they are always just a configuration variant of a physical theme with the exact same fields. And the name field of those themes is user input, which will lead to various magical names. I would leave it to the solution I already implemented below to use the original technical name of the physical theme for database copies.